### PR TITLE
Configuring the sitemap

### PR DESCRIPTION
**Pull request by pomek**
Fixes: #50, #17, #7, #12, #19, #36.

I've updated the tests files in `ckeditor5-basic-styles` and `ckeditor5-typing`. All you have to do is pull the latest changes.

Don't forget about check out to proper branch in `ckeditor5-dev-docs`.

### CODE REVIEW

File: src/middlewares/find-page-title.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * @returns {HtmlFile}
+ */
+module.exports = function findPageTitle( file, next ) {
+	const matchedTitle = file.content.match( /^#(.*)/ );
+
+	if ( !matchedTitle ) {
+		throw new Error( 'Cannot find a title of the page (row beginning with #).' );
+	}
+
+	file.setMetaTag( 'title', matchedTitle[ 1 ].trim() );
```
Comment (pomek) [inline src/middlewares/find-page-title.js]: I guess when the title has found, he should be removed from the content.

@Reinmar, could you confirm?

Comment (Reinmar) [inline src/middlewares/find-page-title.js]: Yes, of course. Now it's displayed twice ;/

**Comment by Reinmar** [comment]
I'll write here all the issues that I noticed. Many unrelated to this PR, but I don't have time now to open new issues for them. We'll have to work on 

* Headings are duplicated (https://github.com/cksource/docs-builder/pull/69#pullrequestreview-1626155).
* There are no headings on the grid boxes: http://localhost/ckeditor5/ckeditor5/build/docs/quick-references.html
* Theme templates werent refreshed. I mean... they became quite unreliable, because there's a big difference in what you get out of them and when doing a real build. For example, ToC doesn't look the same. The test documentation should, preferably, be generated out of real input, not some mocks, which quickly get outdated.
* Clicking on the tabs in samples (HTML, CSS, Result) scrolls to the top.
* The links in ToC should be set active when scrolling the content.
* The left menu looks really bad... Even things like – an active item don't get text decoration on hover (which is should, because it's still a link). But in general, we need to redesign it from scratch.
* When clicking on links in the ToC the content is scrolled in a way that the heading is super close to the top bar. This makes it hardly visible. Headings should have top padding.
* ToC styles are rather poor – items should have left margin, so the structure of the ToC is better visible. Besides, items are too close to each other.
* Headings in the content should have an "anchor" icon on the left (when hovering them). The anchor, should be a link to that heading.
* The parent category name is missing from the page heading (it was on the initial designs).
* I don't like that you need to specify the repo URL in the `docs/config.json`. This should be taken out from the package.json and if not present, then there should be no "Edit on GH" links.
* I found a lot of mixed tabs and spaces in pug files and HTML templates in `dev/`.
* I think that in `config.json` it will be nice to specify full path to the guide file (with extension), because now it's a bit unclear what it is and how to read this.
* The top option in the `config.json` shouldn't be `files` but something like `pages`. This should be clearer.

Fortunately, only just a few items from this list concern this PR. So we should be fine, after you take care of them.

**Comment by pomek** [comment]
I will extract your notes to separate issues (if no exist). Ofc notes related to `docs-builder`, not to this issue.

---

> Theme templates werent refreshed. I mean... they became quite unreliable, because there's a big difference in what you get out of them and when doing a real build. For example, ToC doesn't look the same. The test documentation should, preferably, be generated out of real input, not some mocks, which quickly get outdated.

=> https://github.com/cksource/docs-builder/issues/70

> The links in ToC should be set active when scrolling the content.

=> https://github.com/cksource/docs-builder/issues/35

> The left menu looks really bad... Even things like – an active item don't get text decoration on hover (which is should, because it's still a link). But in general, we need to redesign it from scratch.

=> https://github.com/cksource/docs-builder/issues/64

> When clicking on links in the ToC the content is scrolled in a way that the heading is super close to the top bar. This makes it hardly visible. Headings should have top padding.

=> https://github.com/cksource/docs-builder/issues/61

> ToC styles are rather poor – items should have left margin, so the structure of the ToC is better visible. Besides, items are too close to each other.

=> https://github.com/cksource/docs-builder/issues/66

> Headings in the content should have an "anchor" icon on the left (when hovering them). The anchor, should be a link to that heading.

=> https://github.com/cksource/docs-builder/issues/42

> I found a lot of mixed tabs and spaces in pug files and HTML templates in `dev/`.

=> https://github.com/cksource/docs-builder/issues/68

---

The rest of your notices will be fixed soon.

**Comment by pomek** [comment]
> I don't like that you need to specify the repo URL in the docs/config.json. This should be taken out from the package.json and if not present, then there should be no "Edit on GH" links.

=> https://github.com/cksource/docs-builder/issues/22

**Comment by pomek** [comment]
> The parent category name is missing from the page heading (it was on the initial designs).

=> https://github.com/cksource/docs-builder/issues/23

**Comment by Reinmar** [comment]
Thanks for taking care of all the issues I mentioned. Great job!

**Pull request merged**